server/models/articles.js

```diff
@@ -127,6 +128,46 @@ const getArticleModel = (sequelize, DataTypes) => {
       onUpdate: "CASCADE"
     });
   };
+
+  /**
+   * @description Find or create a new instance of Tags
+   *
+   * @param {Array} tags array of tags
+   * @param {object} tagModel the tag model
+   * @returns {Promise} a promise that resolves to an array of object
+   * @method saveTags
+   */
+  Article.prototype.saveTags = async function(tags) {
+    if (!tags) return;
+    try {
+      let tagInstance;
+      /**
+       * We will loop through the tag list to check
+       */
+      const tagIdArray = await mapAsync(tags, async tag => {
+        tag = tag.trim();
+        tagInstance = await models.Tags.findOne({
```

Comment:
This could have all been prevented using an array of tags in the model but we just have to move on 😄